### PR TITLE
CASMHMS-6402: Prevent unnecessary retries

### PR DESCRIPTION
**Pull request by jwlv**
### Summary and Scope

The `DoHTTPAction()` function was previously falling back to performing an insecure request if the call to `Do()` in hms-certs fails.  The problem with that is that hms-certs already attempts an insecure request if the secure request fails.  Thus, doing this in hms-go-http-lib is redundant and could exacerbate any scaling or resource issues.  This PR removes this second re-attempt.

The ``DoHTTPAction()` function previously did not always drain the response body in all edge cases before closing the response body.  This PR ensures the response body is always drained before closing.

Updated unit test to request skipping verification of self-signed certs.  The internal functionality is not appropriately tested without this change.

Updated Go from v1.23 to v1.24.

Updated module dependencies to latest versions for security updates.

Adopted app version 1.7.0 for CSM 1.7.0 (no helm chart).

### Issues and Related PRs

* Resolves [CASMHMS-6402](https://jira-pro.it.hpe.com:8443/browse/CASMHMS-6402)

### Testing

Tested on:

* `gamora`
* `mug`

Test description:

Tested by pulling the changes here into RTS and deploying RTS on various systems:

- Tested cray-hms-rts on `gamora` to verify no issues communicating with iPDUs via JAWS.
- Tested cray-hms-rts-snmp on `mug` to verify no issues communicating with switches via SNMP.

Watched RTS logs to ensure no deviations from running without these changes.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -233,20 +229,8 @@ func (request *HTTPRequest) DoHTTPAction() (payloadBytes []byte, responseStatusC
...
-	if doErr != nil {
-		//If we tried the secure client and it got an error, try the insecure one
-		if (request.TLSClientPair != nil) &&
-			(request.TLSClientPair.SecureClient != nil) &&
-			(request.TLSClientPair.SecureClient != request.TLSClientPair.InsecureClient) {
-			log.Printf("INFO: Cert-validating HTTP client request failed, trying non-validating client.")
-			request.Client = request.TLSClientPair.InsecureClient
-			resp, doErr = request.Client.Do(req)
-		}
-	}
```
Comment (shunr-hpe) [inline http.go:245]: SMD also uses hms-go-http-lib's DoHTTPAction, and maybe other hms projects do too.
Do you know if SMD still works with this?

**Review by shunr-hpe — APPROVED** [review]

**Review by mbuchmann-hpe — APPROVED** [review]
